libbfddp/bfddp_extra.h

```diff
@@ -0,0 +1,497 @@
+ *
+ * The application must call `bfddp_session_rx_timeout` if the timer expired.
+ *
+ * \param bs the BFD session.
+ * \param arg application argument.
+ */
+typedef void (*bfddp_rx_control_update_cb)(struct bfd_session *bs, void *arg);
+
+/**
+ * Stop session receive control packet timer.
+ *
+ * This is called when the BFD protocol state machine wants to stop receiving
+ * `bfddp_session_rx_timeout` notifications.
+ *
+ * \param bs the BFD session.
+ * \param arg application argument.
+ */
+typedef void (*bfddp_rx_control_stop_cb)(struct bfd_session *bs, void *arg);
+
+/**
+ * BFD state change: it is called every time the session state changes.
+ *
+ * \param bs the BFD session.
+ * \param ostate the old BFD session state.
+ * \param nstate the new BFD session state.
+ */
+typedef void (*bfddp_state_change_cb)(struct bfd_session *bs, void *arg,
+				      enum bfd_state_value ostate,
+				      enum bfd_state_value nstate);
+
+/** The BFD data plane callbacks.  */
+struct bfddp_callbacks {
+	/** Optional callback. */
+	bfddp_session_new_cb bc_session_new;
+	/** Optional callback. */
+	bfddp_session_update_cb bc_session_update;
+	/** Optional callback. */
+	bfddp_session_free_cb bc_session_free;
+
+	/** Mandatory callback. */
+	bfddp_tx_control_cb bc_tx_control;
+	/** Mandatory callback. */
+	bfddp_tx_control_update_cb bc_tx_control_update;
+	/** Mandatory callback. */
+	bfddp_tx_control_stop_cb bc_tx_control_stop;
+
+	/** Mandatory callback. */
+	bfddp_rx_control_update_cb bc_rx_control_update;
+	/** Mandatory callback. */
+	bfddp_rx_control_stop_cb bc_rx_control_stop;
+
+	/** Optional callback. */
+	bfddp_state_change_cb bc_state_change;
+};
+
+/**
+ * Set the library callbacks.
+ *
+ * \param bc the callbacks structure filled.
+ */
+void bfddp_initialize(struct bfddp_callbacks *bc);
+
+
+/*
+ * BFD implementation functions.
+ */
+
+/**
+ * Allocates a new session data structure with the information that came from
+ * FRR's BFD daemon.
+ *
+ * \param bds the BFD message.
+ * \param arg application argument.
+ * \param bctx the FRR BFD data plane to talk to.
+ */
+struct bfd_session *bfddp_session_new(struct bfddp_ctx *bctx, void *arg,
+				      const struct bfddp_session *bm);
+
+/**
+ * Allocates a new session data structure with the information that came from
+ * FRR's BFD daemon.
+ *
+ * \param bs the BFD session.
+ * \param arg application argument.
+ * \param bds the BFD message.
+ */
+void bfddp_session_update(struct bfd_session *bs, void *arg,
+			  const struct bfddp_session *bds);
+
+/**
+ * Allocates a new session data structure with the information that came from
+ * FRR's BFD daemon.
+ *
+ * \param bs the BFD session.
+ */
+void bfddp_session_free(struct bfd_session **bs, void *arg);
+
+/**
+ * Fill the BFD control packet with the information present in the session data
+ * structure.
+ *
+ * \param bs the BFD session.
+ * \param bcp the control packet buffer.
+ */
+void
+bfddp_fill_control_packet(const struct bfd_session *bs,
```

Comment:
There is a mixture of code putting the return type in the same line or with a breakline before the function name. Might be worth homogenizing

Reply:
Nice catch, you are correct... I forgot to run `clang-format`.